addons/addon-base-raas/packages/base-raas-services/lib/data-source/__tests__/environment-resource-service.test.js

```diff
@@ -0,0 +1,645 @@
+          Action: ['s3:GetObject'],
+          Resource: ['arn:aws:s3:::study-bucket/studies/users/user1/Study2/*'],
+        },
+        {
+          Sid: 'List:studies/users/user1/Study2/',
+          Effect: 'Allow',
+          Principal: {
+            AWS: ['arn:aws:iam::accountId1:root'],
+          },
+          Action: ['s3:ListBucket'],
+          Resource: 'arn:aws:s3:::study-bucket',
+          Condition: {
+            StringLike: {
+              's3:prefix': ['studies/users/user1/Study2/*'],
+            },
+          },
+        },
+      ],
+    };
+  };
+
+  const statementSortFn = (statement1, statement2) => {
+    if (statement1.Sid < statement2.Sid) {
+      return -1;
+    }
+    if (statement1.Sid > statement2.Sid) {
+      return 1;
+    }
+    return 0;
+  };
+
+  beforeEach(async () => {
+    // Initialize services container and register dependencies
+    const container = new ServicesContainer();
+    container.register('log', new Logger());
+    container.register('aws', new AwsService());
+    container.register('auditWriterService', new AuditWriterService());
+    container.register('jsonSchemaValidationService', new JsonSchemaValidationService());
+    container.register('lockService', new LockService());
+    container.register('resourceUsageService', new ResourceUsageService());
+    container.register('environmentResourceService', new EnvironmentResourceService());
+    container.register('dbService', new DBService());
+    container.register('settings', new SettingsService());
+
+    await container.initServices();
+
+    environmentResourceService = await container.find('environmentResourceService');
+    const aws = await environmentResourceService.service('aws');
+    AWSMock.setSDKInstance(aws.sdk);
+    environmentResourceService._settings = {
+      get: settingName => {
+        if (settingName === 'studyDataBucketName') {
+          return 'study-bucket';
+        }
+        if (settingName === 'studyDataKmsPolicyWorkspaceSid') {
+          return 'KMS Policy';
+        }
+        if (settingName === 'studyDataKmsKeyArn') {
+          return 'studyKmsKeyAlias';
+        }
+        return undefined;
+      },
+    };
+  });
+
+  afterEach(() => {
+    AWSMock.restore();
+  });
+
+  describe('addToBucketPolicy', () => {
+    it('add new studies to an empty bucket policy', async () => {
+      const studies = testStudiesFn();
+      const s3Policy = testS3PolicyFn();
+      AWSMock.mock('S3', 'getBucketPolicy', (params, callback) => {
+        expect(params).toMatchObject({
+          Bucket: 'study-bucket',
+        });
+        callback(null, { Policy: '{}' });
+      });
+
+      AWSMock.mock('S3', 'putBucketPolicy', (params, callback) => {
+        expect(params).toMatchObject({
+          Bucket: 'study-bucket',
+          Policy: JSON.stringify(s3Policy),
+        });
+        callback(null, {});
+      });
+      await environmentResourceService.addToBucketPolicy({}, studies, 'accountId1');
```

Comment:
You might need to add a line at the end using expect.assertions(number) https://jestjs.io/docs/en/expect#expectassertionsnumber  otherwise if the putBucketPolicy() and getBuckPolicy() are never called, the test will still succeed.

Reply:
Thanks for bringing this up. I have added the assertions for putBucketPolicy. For getBucketPolicy it was implicit since we use the return value later.